agent/acs/handler/attach_eni_handler.go

```diff
@@ -36,11 +36,11 @@ type attachENIHandler struct {
 	cluster           *string
 	containerInstance *string
 	acsClient         wsclient.ClientServer
-	taskEngine        engine.TaskEngine
+	state             dockerstate.TaskEngineState
 }
 
 // newAttachENIHandler returns an instance of the attachENIHandler struct
-func newAttachENIHandler(ctx context.Context, cluster string, containerInstanceArn string, acsClient wsclient.ClientServer, taskEngine engine.TaskEngine, saver statemanager.Saver) attachENIHandler {
+func newAttachENIHandler(ctx context.Context, cluster string, containerInstanceArn string, acsClient wsclient.ClientServer, taskEngineState dockerstate.TaskEngineState, saver statemanager.Saver) attachENIHandler {
```

Comment:
minor: can you also split this into multiple lines while you're here so that it's more readable?